### PR TITLE
#834F & #835F Final decision review page

### PR DESCRIPTION
**Pull request by nmadruga**
### Linked PR:
- Based on PR #453 
- Front-end PR https://github.com/openmsupply/application-manager-web-app/pull/843

### Changes
- Add `is_final_decision` to `review_assignment` and `review`
- Action `generateReviewAssignment` sets reviewAssignment with new field `isFinalDecision` when that is defined in the  `template_permission` (with `canMakeFinalDecision`) for the review that can be assigned this review stage/level.

### CODE REVIEW

**Review by andreievg — APPROVED** [review]
Looks good, just a small suggestion for types.

Also (probably not so important atm, but something to flag) do you think we should change this to be all reviews (apart from current one), regardless of level ? (so that if for some reason there are two deciders, first one to make a decision would lock the other one from making a decision) ?

https://github.com/openmsupply/application-manager-server/blob/e913f8750ea898bb72bbf590e501ec23be1a55a7/plugins/action_update_review_status/src/updateReviewsStatuses.ts#L91-L93

**Comment by nmadruga** [comment]
> do you think we should change this to be all reviews (apart from current one), regardless of level ? (so that if for some reason there are two deciders, first one to make a decision would lock the other one from making a decision) ?

There is still one front-end issue to deal with locking other `final decision` users in the same level https://github.com/openmsupply/application-manager-web-app/issues/836, when this can be done. Although I'm not sure about `to be all reviews (apart from current one)` because logic is important for other types of reviews (with different levels)

File: plugins/action_generate_review_assignment_records/src/databaseMethods.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ const databaseMethods = (DBConnect: any) => ({
         allowedSections,
         levelNumber,
         isLastLevel,
+        isFinalDecision,
```
[the remark this answers is not in the document]
Comment (nmadruga) [reply plugins/action_generate_review_assignment_records/src/databaseMethods.ts:44]: Done: 9923dab

**Comment by CarlosNZ** [comment]
Quick comment -- I notice after making a "final decision" the application still shows as PENDING. Probably needs a "Change Outcome" action, or if there is one, it isn't working correctly.

**Comment by nmadruga** [comment]
> Quick comment -- I notice after making a "final decision" the application still shows as PENDING. Probably needs a "Change Outcome" action, or if there is one, it isn't working correctly.

Having a look on this problem now.

**Comment by nmadruga** [comment]
@CarlosNZ problem fixed. Thanks for pointing it out. I should have tested the PR where that was included again. Ii had it working but it stopped working after a new addition was done. Now it should be all good to test.

**Pull request merged**